src/gui/tray/ActivityItem.qml

```diff
@@ -0,0 +1,265 @@
+                spacing: 4
+                Layout.alignment: Qt.AlignLeft
+                
+                Text {
+                    id: activityTextTitle
+                    text: (type === "Activity" || type === "Notification") ? subject : message
+                    width: parent.width
+                    elide: Text.ElideRight
+                    font.pixelSize: Style.topLinePixelSize
+                    color: activityTextTitleColor
+                }
+                
+                Text {
+                    id: activityTextInfo
+                    text: (type === "Sync") ? displayPath
+                                            : (type === "File") ? subject
+                                                                : (type === "Notification") ? message
+                                                                                            : ""
+                    height: (text === "") ? 0 : activityTextTitle.height
+                    width: parent.width
+                    elide: Text.ElideRight
+                    font.pixelSize: Style.subLinePixelSize
+                }
+                
+                Text {
+                    id: activityTextDateTime
+                    text: dateTime
+                    height: (text === "") ? 0 : activityTextTitle.height
+                    width: parent.width
+                    elide: Text.ElideRight
+                    font.pixelSize: Style.subLinePixelSize
+                    color: "#808080"
+                }
+            }
+            
+            RowLayout {
+                id: activityActionsLayout
+                spacing: 0
+                Layout.alignment: Qt.AlignHCenter | Qt.AlignVCenter
+                Layout.minimumWidth: 28
+                Layout.fillWidth: true
+                
+                function actionButtonIcon(actionIndex) {
+                    const verb = String(model.links[actionIndex].verb);
+                    if (verb === "WEB" && (model.objectType === "chat" || model.objectType === "call")) {
+                        return "qrc:///client/theme/reply.svg";
+                    } else if (verb === "DELETE") {
+                        return "qrc:///client/theme/close.svg";
+                    }
+                    
+                    return "qrc:///client/theme/confirm.svg";
+                }
+                
+                Repeater {
+                    model: activityItem.links.length > activityListView.maxActionButtons ? 1 : activityItem.links.length
+                    
+                    ActivityActionButton {
+                        id: activityActionButton
+                        
+                        readonly property int actionIndex: model.index
+                        readonly property bool primary: model.index === 0 && String(activityItem.links[actionIndex].verb) !== "DELETE"
+                        
+                        height: activityItem.height
```

Comment:
`Layout.fillHeight` otherwise we need to hunt for `activityItem` defined long before we are here